src/sentry/api/event_search.py

```diff
@@ -558,7 +557,18 @@ def visit_boolean_operator(self, node, children):
         return node.text
 
     def visit_value(self, node, children):
-        return node.text
+        # A properly quoted value will match the quoted value regex, so any unescaped
+        # quotes are errors.
+        for (idx, c) in enumerate(node.text):
```

Comment:
Do you need to enumerate all the characters for this check? It seems like something you could do with a regular expression.

Reply:
I don't know that a regex would be more performant than enumerate but I can check.

Reply:
Well normally I would say if you solve a problem with regex then you have two problems. After testing though, the regex option is noticeably faster than the non-regex version. Thanks Python.

Reply:
Probably significantly fewer allocations being made.